sdk/all/src/test/java/io/opentelemetry/sdk/ScopeConfiguratorTest.java

```diff
@@ -0,0 +1,217 @@
+                    .build())
+            .setLoggerProvider(
+                SdkLoggerProvider.builder()
+                    .addLogRecordProcessor(SimpleLogRecordProcessor.create(logRecordExporter))
+                    .addLoggerConfiguratorCondition(nameEquals("scopeB"), LoggerConfig.disabled())
+                    .build())
+            .build();
+
+    simulateInstrumentation(sdk);
+
+    // Collect all the telemetry. Ensure we don't see any from scopeB, and that the telemetry from
+    // scopeA and scopeC is valid.
+    assertThat(spanExporter.getFinishedSpanItems())
+        .satisfies(
+            spans -> {
+              Map<InstrumentationScopeInfo, List<SpanData>> spansByScope =
+                  spans.stream()
+                      .collect(Collectors.groupingBy(SpanData::getInstrumentationScopeInfo));
+              assertThat(spansByScope.get(InstrumentationScopeInfo.create("scopeA"))).hasSize(1);
+              assertThat(spansByScope.get(InstrumentationScopeInfo.create("scopeB"))).isNull();
+              assertThat(spansByScope.get(InstrumentationScopeInfo.create("scopeC"))).hasSize(1);
+            });
+    assertThat(metricReader.collectAllMetrics())
+        .satisfies(
+            metrics -> {
+              Map<InstrumentationScopeInfo, List<MetricData>> metricsByScope =
+                  metrics.stream()
+                      .collect(Collectors.groupingBy(MetricData::getInstrumentationScopeInfo));
+              assertThat(metricsByScope.get(InstrumentationScopeInfo.create("scopeA"))).hasSize(1);
+              assertThat(metricsByScope.get(InstrumentationScopeInfo.create("scopeB"))).isNull();
+              assertThat(metricsByScope.get(InstrumentationScopeInfo.create("scopeC"))).hasSize(1);
+            });
+    assertThat(logRecordExporter.getFinishedLogRecordItems())
+        .satisfies(
+            logs -> {
+              Map<InstrumentationScopeInfo, List<LogRecordData>> logsByScope =
+                  logs.stream()
+                      .collect(Collectors.groupingBy(LogRecordData::getInstrumentationScopeInfo));
+              assertThat(logsByScope.get(InstrumentationScopeInfo.create("scopeA"))).hasSize(1);
+              assertThat(logsByScope.get(InstrumentationScopeInfo.create("scopeB"))).isNull();
+              assertThat(logsByScope.get(InstrumentationScopeInfo.create("scopeC"))).hasSize(1);
+            });
+  }
+
+  /** Disable all scopes by default and enable a single scope. */
+  @Test
+  void disableAllScopesExceptB() {
+    OpenTelemetrySdk sdk =
+        OpenTelemetrySdk.builder()
+            .setTracerProvider(
+                SdkTracerProvider.builder()
+                    .addSpanProcessor(SimpleSpanProcessor.create(spanExporter))
+                    .setTracerConfigurator(
+                        TracerConfig.configuratorBuilder()
+                            .setDefault(TracerConfig.disabled())
```

Comment:
So, reading through and comparing with the above test, this seems to exhibit specific knowledge about the default behavior, while the other test above does not. A reader might wonder why this one sets the default but the other one doesn't. 

I think this is a perfectly reasonable default, but it makes the tests read a little differently. Maybe setting the default in the above test could help. This is a nit for sure.

Reply:
Its a good question.

The default of having scopes enabled by default is required for backwards compatibility, so we're stuck with that.

The test above benefits from this, and is able to express simpler rules since it only needs to express which scopes are disabled.

This test changes the default from enabled to disabled, and opts back in to specific instrumentations which are enabled. Changing the defaults requires the caller to express things a little differently / more verbosely.

So if by having the tests express the rules differently, we can show users how to very quickly express the simple / common case of opting out of specific scopes, but loose symmetry. If we the tests, then we have symmetry but no longer demonstrate the simple / common case. Tradeoffs...